define/types.go

```diff
@@ -178,11 +178,16 @@ func TempDirForURL(dir, prefix, url string) (name string, subdir string, err err
 	return "", "", errors.New("unreachable code reached")
 }
 
-func cloneToDirectory(url, dir string) ([]byte, string, error) {
+// parseGitBuildContext parses git build context to `repo`, `sub-dir`
+// `branch/commit`, accepts GitBuildContext in the format of
+// `repourl.git#branch/commit:subdir`.
+func parseGitBuildContext(url string) (string, string, string) {
```

Comment:
Does this need to be its own function any more, or can it be inlined back into `cloneToDirectory()`?

Reply:
Just moved it out so unit tests can be added for parsing.